_api_app/app/Sites/SitesDataService.php

```diff
@@ -123,16 +124,35 @@ public function create($cloneFrom = null)
         $name = 'untitled-' . uniqid();
         $dir = $this->XML_SITES_ROOT . '/' . $name;
 
-        @mkdir($dir, 0777, true);
-
         if ($cloneFrom != null) {
             $src = $cloneFrom == '0' ? $this->XML_MAIN_ROOT : $this->XML_SITES_ROOT . '/' . $cloneFrom;
+            $name = 'copy-of-'.$cloneFrom;
+            if($cloneFrom == '0') {
+                $title = $this->MAIN_SITE_DEFAULT_TITLE;
+            } else {
+                foreach ($sites as $site) {
+                    if ($site['name'] === $cloneFrom) {
+                        $title = $site['title'];
+                        break;
+                    }
+                }
+            }
+            $title = 'Copy of '.$title;
+            $copyName = $name;
+            $i=1;
+            while (file_exists($this->XML_SITES_ROOT . '/' . $name)) {
```

Comment:
It would be faster to read the whole directory and then go through it with `foreach`. Since there can be many sites and asking OS if the file exists is relatively slow.